src/custom-element.js

```diff
@@ -1429,9 +1440,44 @@ function createBaseAmpElementProto(win) {
 export function registerElement(win, name, implementationClass) {
   knownElements[name] = implementationClass;
 
-  win.document.registerElement(name, {
-    prototype: createAmpElementProto(win, name),
-  });
+  const supportsCustomElementsV1 = 'customElements' in win;
+  if (supportsCustomElementsV1) {
+    // `customElements.define` requires a function or ES2015 class to be passed
```

Comment:
This comment may be unclear -- I think that's already happening here.
- Both V0 and V1 elements inherit from `HTMLElement`.
- API for V0 expects a prototype object whereas V1 expects a function/class.
- There's a Babel issue with `class extends HTMLElement`, so we manually transpile instead